nearcache/nearcache_it_test.go

```diff
@@ -0,0 +1,329 @@
+			f: func(ctx context.Context, tcx it.MapTestContext, i int64) {
+				if err := tcx.M.Delete(ctx, i); err != nil {
+					tcx.T.Fatal(err)
+				}
+			},
+		},
+	}
+	invalidationRunner(t, testCases)
+}
+
+func TestAfterPutNearCacheIsInvalidated(t *testing.T) {
+	// port of: com.hazelcast.client.map.impl.nearcache.ClientMapNearCacheTest#testAfterPutAsyncNearCacheIsInvalidated
+	testCases := []mapTestCase{
+		{
+			name: "Put",
+			f: func(ctx context.Context, tcx it.MapTestContext, i int64) {
+				v, err := tcx.M.Put(ctx, i, i)
+				if err != nil {
+					tcx.T.Fatal(err)
+				}
+				require.Equal(t, i, v)
+			},
+		},
+	}
+	invalidationRunner(t, testCases)
+}
+
+func TestAfterSetNearCacheIsInvalidated(t *testing.T) {
+	// port of: com.hazelcast.client.map.impl.nearcache.ClientMapNearCacheTest#testAfterSetAsyncNearCacheIsInvalidated
+	testCases := []mapTestCase{
+		{
+			name: "Set",
+			f: func(ctx context.Context, tcx it.MapTestContext, i int64) {
+				if err := tcx.M.Set(ctx, i, i); err != nil {
+					tcx.T.Fatal(err)
+				}
+			},
+		},
+		{
+			name: "SetWithTTL",
+			f: func(ctx context.Context, tcx it.MapTestContext, i int64) {
+				if err := tcx.M.SetWithTTL(ctx, i, i, 1*time.Second); err != nil {
+					tcx.T.Fatal(err)
+				}
+			},
+		},
+		{
+			name: "SetWithTTLAndMaxIdle",
+			f: func(ctx context.Context, tcx it.MapTestContext, i int64) {
+				if err := tcx.M.SetWithTTLAndMaxIdle(ctx, i, i, 1*time.Second, 2*time.Second); err != nil {
+					tcx.T.Fatal(err)
+				}
+			},
+		},
+	}
+	invalidationRunner(t, testCases)
+}
+
+func TestAfterTryRemoveNearCacheIsInvalidated(t *testing.T) {
+	// port of: com.hazelcast.client.map.impl.nearcache.ClientMapNearCacheTest#testAfterTryRemoveNearCacheIsInvalidated
+	testCases := []mapTestCase{
+		{
+			name: "TryRemove",
+			f: func(ctx context.Context, tcx it.MapTestContext, i int64) {
+				v, err := tcx.M.TryRemove(ctx, i)
+				if err != nil {
+					tcx.T.Fatal(err)
+				}
+				require.True(t, v.(bool))
+			},
+		},
+		{
+			name: "TryRemoveWithTimeout",
+			f: func(ctx context.Context, tcx it.MapTestContext, i int64) {
+				v, err := tcx.M.TryRemoveWithTimeout(ctx, i, 10*time.Second)
+				if err != nil {
+					tcx.T.Fatal(err)
+				}
+				require.True(t, v.(bool))
+			},
+		},
+	}
+	invalidationRunner(t, testCases)
+}
+
+func TestAfterTryPutNearCacheIsInvalidated(t *testing.T) {
+	// port of: com.hazelcast.client.map.impl.nearcache.ClientMapNearCacheTest#testAfterTryPutNearCacheIsInvalidated
+	testCases := []mapTestCase{
+		{
+			name: "Put",
+			f: func(ctx context.Context, tcx it.MapTestContext, i int64) {
+				v, err := tcx.M.Put(ctx, i, i)
+				if err != nil {
+					tcx.T.Fatal(err)
+				}
+				require.Equal(tcx.T, i, v)
+			},
+		},
+		{
+			name: "PutWithMaxIdle",
+			f: func(ctx context.Context, tcx it.MapTestContext, i int64) {
+				v, err := tcx.M.PutWithMaxIdle(ctx, i, i, 10*time.Minute)
+				if err != nil {
+					tcx.T.Fatal(err)
+				}
+				require.Equal(tcx.T, i, v)
+			},
```

Comment:
```suggestion
			name: "TryPut",
			f: func(ctx context.Context, tcx it.MapTestContext, i int64) {
				v, err := tcx.M.TryPut(ctx, i, i)
				if err != nil {
					tcx.T.Fatal(err)
				}
				require.Equal(tcx.T, i, v)
			},
		},
		{
			name: "TryPutWithTimeout",
			f: func(ctx context.Context, tcx it.MapTestContext, i int64) {
				v, err := tcx.M.TryPutWithTimeout(ctx, i, i, 10*time.Second)
				if err != nil {
					tcx.T.Fatal(err)
				}
				require.Equal(tcx.T, i, v)
			},
```

Reply:
It seems I didn't add nearcache support for TryPut, but added a wrong test for it. Thanks, added at: https://github.com/hazelcast/hazelcast-go-client/pull/806/commits/de818d94eb13ae503e6ecfa8f8f0723879d22e38